src/features/world/ui/NPCModals.tsx

```diff
@@ -184,6 +131,26 @@ export const NPCModals: React.FC<Props> = ({ onClose, onOpen }) => {
             </div>
           </CloseButtonPanel>
         )}
+        {/* Delivery NPC's */}
+        {npc === "pumpkin' pete" && (
+          <DeliveryModal npc={npc} onClose={closeModal} />
+        )}
+        {npc === "blacksmith" && (
+          <DeliveryModal npc={npc} onClose={closeModal} />
+        )}
+        {npc === "raven" && <DeliveryModal npc={npc} onClose={closeModal} />}
+        {npc === "tywin" && <DeliveryModal npc={npc} onClose={closeModal} />}
+        {npc === "grimbly" && <DeliveryModal npc={npc} onClose={closeModal} />}
+        {npc === "grimtooth" && (
+          <DeliveryModal npc={npc} onClose={closeModal} />
+        )}
+        {npc === "bert" && <DeliveryModal npc={npc} onClose={closeModal} />}
+        {npc === "timmy" && <DeliveryModal npc={npc} onClose={closeModal} />}
+        {npc === "old salty" && (
+          <DeliveryModal npc={npc} onClose={closeModal} />
+        )}
+        {npc === "betty" && <DeliveryModal npc={npc} onClose={closeModal} />}
+        {npc === "cornwell" && <DeliveryModal npc={npc} onClose={closeModal} />}
       </Modal>
```

Comment:
Unfortunately I couldn't fulfil Grubnuk's delivery (it only showed the Community Island modal)

Similar to the Blacksmith, we may need some mechanism to choose whether we want to talk/deliver or do their normal action